api/index.js

```diff
@@ -1,27 +1,47 @@
-import express from 'express';
-import path from 'path';
-import logger from 'morgan';
-import bodyParser from 'body-parser';
-import routes from './routes/index';
-import errors from './routes/error-handling';
-import compression from 'compression';
+import Hapi from 'hapi';
+import { register as routes } from './routes/index';
+/* import path from 'path';
+import errors from './routes/error-handling';*/
 
-const app = express();
+const server = new Hapi.Server();
 
-// this will be used to tell the server when to stop taking connections
-app.set('shutting-down', false);
-app.use(compression());
-app.use(logger('dev'));
-app.use((req, res, next) => {
-  // Timeout connection so the client tries to reconnect once the server is back up
-  if (app.settings['shutting-down']) req.connection.setTimeout(1);
-  return next();
-});
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(express.static(path.join(__dirname, 'public')));
-app.use(routes);
-app.use(errors);
+export function createServer({ port }) {
+  server.connection({
+    port,
+    host: '0.0.0.0',
+    router: {
+      stripTrailingSlash: true,
+    },
+  });
 
+  server.register([
```

Comment:
Pulling this into another file/module doesn't feel right yet. I would like to see how this works out first. Unless you have other ideas.